hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/ipc/TestIdentityProviders.java

```diff
@@ -71,12 +73,20 @@ public void testPluggableIdentityProvider() {
   @Test
   public void testUserIdentityProvider() throws IOException {
     UserIdentityProvider uip = new UserIdentityProvider();
-    String identity = uip.makeIdentity(new FakeSchedulable());
+    FakeSchedulable fakeSchedulable = new FakeSchedulable();
+    String identity = uip.makeIdentity(fakeSchedulable);
 
     // Get our username
     UserGroupInformation ugi = UserGroupInformation.getCurrentUser();
     String username = ugi.getUserName();
 
     assertEquals(username, identity);
+
+    // FakeSchedulable doesn't override getCallerContext()
+    // accessing it should throw a NotImplementedException
+    NotImplementedException nie = assertThrows(
```

Comment:
our own LambdaTestUtils.intercept() does this with better reporting of mismatch, including rethrowing the failing exception if there's no string match

Reply:
Thanks for letting me know. I'll use that instead.